src/Parse.y

```diff
@@ -1,16 +1,23 @@
+{-
+
+  this is a happy grammar :)
+
+ -}
+
 {
+
 module Parse where
 import Prelude hiding (EQ, GT, LT)
 import Lex
 import EVM.ABI
 import EVM.Solidity (SlotType(..))
 import qualified Data.List.NonEmpty as NonEmpty
 import Syntax
-import ErrM
+
 }
 
 %name parse
-%monad { Err } { (>>=) } { return }
+%monad { Either String } { (>>=) } { return }
```

Comment:
the `Err` monad carries position information and not just a string, which gives us some more options for more comprehensive error messages

Reply:
OK, will revert. I had removed position info temporarily to make it a bit easier to hack the grammar.

Reply:
Actually, why do we need to track position info in the error monad? `parseError` has access to the current position, regardless, by just looking at the remaining list of lexemes.

Reply:
ok, so we could have a function `parseError -> AlexPosn` ?